packages/teleport/src/services/joinToken/joinToken.ts

```diff
@@ -17,35 +17,32 @@ limitations under the License.
 import api from 'teleport/services/api';
 import cfg from 'teleport/config';
 
+import { makeLabelMapOfStrArrs } from '../agents/make';
+
 import makeJoinToken from './makeJoinToken';
-import { JoinToken, JoinMethod, JoinRole, JoinRule } from './types';
+import { JoinToken, JoinRule, JoinTokenRequest } from './types';
 
 class JoinTokenService {
   fetchJoinToken(
-    // roles is a list of join roles, since there can be more than
-    // one role associated with a token.
-    roles: JoinRole[],
-    joinMethod: JoinMethod = 'token',
-    // rules is a list of allow rules associated with the join token
-    // and the node using this token must match one of the rules.
-    rules: JoinRule[] = [],
+    req: JoinTokenRequest,
     signal: AbortSignal = null
   ): Promise<JoinToken> {
     return api
       .post(
         cfg.getJoinTokenUrl(),
         {
-          roles,
-          join_method: joinMethod,
-          allow: makeAllowField(rules),
+          roles: req.roles,
+          join_method: req.method || 'token',
+          allow: makeAllowField(req.rules || []),
+          agent_matcher_labels: makeLabelMapOfStrArrs(req.agentMatcherLabel),
```

Comment:
as `agentMatcherLabel` is optional can you define a default plz.

Reply:
it does provide a default (the `makeLabelMapOfStrArrs`), it's also tested with empty values and comes out as expected